### DISCUSSION TITLE
Vercel ai SDK for User interface

### DISCUSSION THREAD

**Discussion by jamon8888** [lfnovo/open-notebook#20] [Ideas]
Streamlit is cool but for a broader reach, it could be interesting to create to create a great user interface. There is tools like bolt New that an deliver a really good  interface in hours, for deployment of a separated user interface on vercel , people could install it in one click. I could help

**Comment by lfnovo** [comment]
Hi @jamon8888. You are totally right. Streamlit is just a temporary placeholder to develop the first version of the app so that front-end dev doesn't slow me down (I am more of a backend guy). In the meantime, been studying how Surreal works with GraphQL and I am thinking this could be a great way to expose the data to a web client, like you mentioned. Also, the client can call Langgraph APIs exposed via Langserve. 

I have a little more experience with Vue altough not a lot. I see people usually prefer to do things in React. Do you have a preference? I'd love to discuss how you could help as this is a major milestone for improving the user experience. Let

**Comment by lfnovo** [comment]
Closing this as completed and superseded. Open Notebook has since moved away from the original Streamlit UI and now ships a dedicated Next.js frontend.

Thank you @jamon8888 for raising this direction early and offering to help. A richer web interface became an important part of the project.
